### PR TITLE
3.0 update of startup

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/startup?view=aspnetcore-3.0&branch=pr-en-us-13606)

### CODE REVIEW

**Comment by dougbu** [comment]
@Tratcher I don't feel qualified to review these docs. Are you the right person?

File: aspnetcore/fundamentals/startup/3.0_samples/StartupFilterSample/RequestSetOptionsMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        // Test with https://localhost:5001/?option=hello
+        public async Task Invoke(HttpContext httpContext)
+        {
+            var option = httpContext.Request.Query["option"];
+
+            if (!string.IsNullOrWhiteSpace(option))
+            {
+                _injectedOptions.Value.Option = WebUtility.HtmlEncode(option);
```
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/startup/3.0_samples/StartupFilterSample/RequestSetOptionsMiddleware.cs]: Wow, that really simplifies the sample code!

**Comment by Rick-Anderson** [comment]
@Tratcher  address all your input.

File: aspnetcore/fundamentals/startup.md [hunk 2]
```diff
@@ -74,55 +130,89 @@ The [ASP.NET Core templates](/dotnet/core/tools/dotnet-new) configure the pipeli
...
+::: moniker-end
+
+Each `Use` extension method adds one or more middleware components to the request pipeline. For instance, <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles*> configures [middleware](xref:fundamentals/middleware/index) to serve [static files](xref:undamentals/static-files).
 
 Each middleware component in the request pipeline is responsible for invoking the next component in the pipeline or short-circuiting the chain, if appropriate. If short-circuiting doesn't occur along the middleware chain, each middleware has a second chance to process the request before it's sent to the client.
 
-Additional services, such as `IHostingEnvironment` and `ILoggerFactory`, may also be specified in the `Configure` method signature. When specified, additional services are injected if they're available.
+Additional services, such as `IHostingEnvironment` and `ILoggerFactory`, can be specified in the `Configure` method signature. These services are injected if they're available.
```
[the remark this answers is not in the document]
Comment (Tratcher) [reply aspnetcore/fundamentals/startup.md]: "Or anything defined in ConfigureServices" also applies to <3.0

File: aspnetcore/fundamentals/startup.md [hunk 3]
```diff
@@ -36,41 +36,55 @@ The preceding sample is for [Razor Pages](xref:razor-pages/index); the MVC versi
...
 
-[!code-csharp[](startup/3.0_samples/Program3.cs?name=snippet_Program)]
+[!code-csharp[](startup/3.0_samples/Program3.cs?name=snippet_Program&highlight=12)]
+
+The host provides services that are available to the `Startup` class constructor. The app adds additional services via `ConfigureServices`. Both the host and app services are available in `Configure` and throughout the app.
+
+Only the following service types can be injected into the `Startup` constructor when using <xref:Microsoft.Extensions.Hosting.IHostBuilder>:
+
+* `IWebHostEnvironment`
+* `IHostEnvironment`
```
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/startup.md]: Yes, we don't have 3.0 yet. We're working on that.

**Pull request merged**